src/assets/TokenListController.ts

```diff
@@ -17,30 +17,53 @@ const DEFAULT_THRESHOLD = 60 * 30 * 1000;
 
 const name = 'TokenListController';
 
-interface DataCache {
-  timestamp: number;
-  data: Token[];
-}
-interface TokensChainsCache {
-  [chainSlug: string]: DataCache;
+type BaseToken = {
+  name: string;
+  symbol: string;
+  decimals: number;
+};
+
+type StaticToken = {
+  logo: string;
+  erc20: boolean;
+} & BaseToken;
+
+export type ContractMap = {
+  [address: string]: StaticToken;
+};
+
+export type DynamicToken = {
+  address: string;
+  occurrences: number;
+  aggregators: string[];
+  iconUrl: string;
+} & BaseToken;
+
+export enum MediaExtType {
+  SVG = 'SVG',
+  PNG = 'PNG',
+  JPG = 'JPG',
 }
 
-type Token = {
-  name: string;
+export type IconPath = {
+  filePath: string;
+  type: MediaExtType;
+};
+
+export type TokenListToken = {
   address: string;
-  decimals: number;
-  symbol: string;
+  iconUrl: string | null;
+  iconPath: IconPath | null;
```

Comment:
What is the purpose of this property? I see that it has two properties ("filePath" which is the _filename_, not really a path, and "type" which is the file extension), but I don't know what it's for. Especially if `iconUrl` will remain.

I had thought your intention was to normalize `iconUrl` to be a real URL for both static and dynamic tokens, but that no longer appears to be your goal.

Reply:
Removed iconPath